lib/queue.rb

```diff
@@ -1,31 +1,58 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(220)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    elsif @front == @back
+      raise StandardError.new "Cannot add element. Queue is full."
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    element = @store[@front]
+
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+
+    return element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    length = 0
+
+    if @back > @front
+      length = (@back - @front) + 1
+    elsif @front > @back
+      length = @store.length - @front
+      i = 0
+
+      until @store[i] == nil
+        length += 1
+        i += 1
+      end
+    end
+
+    return length
```

Comment:
This works, but in the elsif you could do some arithmetic instead of the loop

```ruby
  else
    # if back is less than front
    # Get the length from the front to the end and then add the length from index 0 to @back 
    length = @store.length - @front + @back
```